### PR TITLE
fprintd: 1.94.2 -> 1.94.3, python312Packages.pypamtest 1.1.3 -> 1.1.5, fix build issues

### PR DESCRIPTION
**Pull request by gador**
## Description of changes

Update and fix of `python312Packages.pypamtest` to fix build issue in `fprintd` (which coulnd't find the python bindings)

Also updates `fprintd` and `pypamtest` 
[Release notes fprintd](https://gitlab.freedesktop.org/libfprint/fprintd/-/releases)
[Changelog pypamtest](https://git.samba.org/?p=pam_wrapper.git;a=log;h=refs/tags/pam_wrapper-1.1.5)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by NickCao** [comment]
Alternative fix: https://gitlab.archlinux.org/archlinux/packaging/packages/pam_wrapper/-/blob/main/pam_wrapper-1.1.5-site-packages.patch

**Comment by gador** [comment]
> Alternative fix: https://gitlab.archlinux.org/archlinux/packaging/packages/pam_wrapper/-/blob/main/pam_wrapper-1.1.5-site-packages.patch

Cool, didn't find that patch when I was looking for a solution 👍 

I'm fine either way. In the end both provide the path for fprintd and fix the build issue on current master

**Comment by NickCao** [comment]
> I'm fine either way. In the end both provide the path for fprintd and fix the build issue on current master

Actually I prefer yours.

File: pkgs/development/libraries/libpam-wrapper/default.nix [hunk 1]
```diff
@@ -1,23 +1,34 @@
...
     rev = "pam_wrapper-${version}";
-    sha256 = "00mqhsashx7njrvxz085d0b88nizhdy7m3x17ip5yhvwsl63km6p";
+    hash = "sha256-AtfkiCUvCxUfll6lOlbMyy5AhS5R2BGF1+ecC1VuwzM=";
   };
 
+  patches = [
+    (substituteAll {
+      src = ./python.patch;
+      siteDir = if enablePython then "${python.sitePackages}" else "";
+      includeDir = if enablePython then "include/${lib.versions.major python.version}.${lib.versions.minor python.version}" else "";
```
Comment (NickCao) [inline pkgs/development/libraries/libpam-wrapper/default.nix:28]:
```suggestion
      siteDir = lib.optionalString enablePython python.sitePackages;
      includeDir = lib.optionalString enablePython "include/${python.libPrefix}";
```

**Comment by NickCao** [comment]
@ofborg build fprintd

**Review by NickCao — APPROVED** [review]
Tested on my laptop.

**Comment by Daniel-McLarty** [comment]
So when can I build my system again and how can I get this to bypass? I removed all of everything relating to fprintd from my config but it still wont build because of this. Can I force it to build from this commit even if the hydra will take another month to fix itself?

**Comment by gador** [comment]
> So when can I build my system again and how can I get this to bypass? I removed all of everything relating to fprintd from my config but it still wont build because of this. Can I force it to build from this commit even if the hydra will take another month to fix itself?

Can you paste your build/error log?

How do you build your system? `configuration.nix` or `flake.nix`?
Do you use unstable?

You could use an overlay with an override of the fix, or (if you use flakes) add an input of the commit and use that as an overlay.
Also, I don't believe it will take that long to reach the unstable channel

**Comment by Daniel-McLarty** [comment]
@gador I am just going to keep my flake.lock from when the system worked and not update till I know the hydra fixed itself. Unrelated, but how is it that major breaking build things get past the hydra anyways? Doesn't it have to build the system and have it work BEFORE it yells out "known good" to all the users?

**Comment by gador** [comment]
Well the major update to python3.12 happens on master (and later on `unstable`). This was communicated by @mweinelt (hexa) on [discourse](https://discourse.nixos.org/t/breaking-changes-announcement-for-unstable/17574/53?u=gador)

The discussion about breaking changes was already [stirred up](https://discourse.nixos.org/t/can-we-please-stop-breaking-stuff-willy-nilly/48496) a couple of days ago.
TL/DR: We need to update and upstream usually lags behind, so we need to fix stuff along the way.

This is `unstable` after all.

These kind of major changes take a long time to build on hydra and it is clear we need to fix a lot, anyway.

> Doesn't it have to build the system and have it work BEFORE it yells out "known good" to all the users?

Only on stable (IMO)

> I am just going to keep my flake.lock from when the system worked 

I do the same. FWIW I keep a fork of nixpkgs and cherry-pick the fixes diretly to it ;-)

**Pull request merged**